packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -1461,7 +1461,7 @@ class Navigator extends StatefulWidget {
     this.onUnknownRoute,
     this.transitionDelegate = const DefaultTransitionDelegate<dynamic>(),
     this.reportsRouteUpdateToEngine = false,
-    this.clipBehavior = Clip.hardEdge,
```

Comment:
If this is a nested Navigator on only part of the screen, any route transition that does paint transform (MaterialPageRoute for example) will be drawn outside of the rect.

I think for most nested navigation scenario you will want this to clip.

Reply:
Isn't this the per overlay clip? Or did I totally misunderstand this code

Reply:
If its just nested navigators then ignore this, I misread the code!